ast/parser.go

```diff
@@ -176,6 +179,12 @@ func (p *Parser) WithSkipRules(skip bool) *Parser {
 	return p
 }
 
+// WithJSONFields sets the JSON fields that should be exposed in the JSON
+func (p *Parser) WithJSONFields(jsonFields map[string]bool) *Parser {
```

Comment:
Not sure if JSONFields would be better as a custom options type. e.g. `ASTJSONMarshalOptions` which then each AST struct could read from to determine how to marshal itself...

Reply:
Hmm.. as opposed to this being global and applied to all nodes?

Reply:
I could have been clearer here. I was considering changing the type of the `jsonFields` from `map[string]bool` to a custom struct for this use case, such as `ASTJSONMarshalOptions`.

`ASTJSONMarshalOptions` could wrap settings for global options and individual types. E.g. allow the changing of the json marshalling behaviour for a particular type only.

Reply:
Hmmm... what would be an example use-case where we'd want to change the marshalling? :thinking:

Reply:
Sometimes we might want to have more control over which fields on which types are enabled / disabled.

For example, we might want to turn on locations for `every` only. This would be marginally easier to retrofit with a custom struct rather than a map[string]bool.

```
map[string]bool{
  "every_location": true,
}
```

```
ASTJSONMarshalOptions{
  EveryLocations: true,
}
```

Mostly because it'd reduce the number of presence checks in the map and make it easier to make targeted adjustments to certain types.

Reply:
Replacing the `map` in `JSONFields map[string]bool` with a custom struct would give us the flexibility to expand the feature. We can always start with the global options and then later add in more fine-grained controls.

Another thought is do we need a map of `node field type`  -> `expose or not`. We can simply not expose by default (ie. current behavior) and then only do so when explicitly set.

Reply:
Ok thanks. I think that makes sense to me too. I'll have a go at this now.

> do we need a map of node field type -> expose or not.

Perhaps it makes sense to do this now as part of an options wrapper which might grow in future. I'll see what it looks like.

Reply:
I've had a go at this in e15a39be, let me know what you think 🙂